### PR TITLE
[IA-3556] Add SecurityGroup Project label

### PR DESCRIPTION
**Pull request by dexamundsen**
Set project label to indicate application allowlist policy (https://broadworkbench.atlassian.net/browse/IA-3556)

- readme.md: link to Broad helm file repo for configPath
- build.grade: update dependencies to resolve build warnings
- java.version: set 11 to 11.0
- Add dsde support for m1 chips (from as https://github.com/DataBiosphere/terra-workspace-manager/pull/696)
- Add securityGroup to GCP config, and subsequently to project label if its is not empty
- Add new toolsalpha/vpc-dev config file with security-group set to high
- Cleaned up pool-schema files (sorted by config name)

Pending - Integration Test update; local testing

### CODE REVIEW

**Review by yonghaoy — APPROVED** [review]
LGTM! Thanks. Can you also help to create a new VPC pool in dev with high security label?

File: build.gradle [hunk 1]
```diff
@@ -208,6 +203,18 @@ swaggerSources {
...
+spotless {
+    java {
+        googleJavaFormat()
+        targetExclude "${buildDir}/**"
+    }
+}
+tasks.withType(com.diffplug.gradle.spotless.SpotlessTask) {
+    it.dependsOn(swaggerSources.server.code)
+    it.dependsOn(swaggerSources.resource.code)
+}
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply build.gradle:216]: reason for warning without changes - 

- swaggerSources.* uses swaggerOutputDir = "${buildDir}/generated"
- spotless declares targetExclude "${buildDir}/**"

hence the dependency

File: src/main/resources/config/toolsalpha/pool_schema.yml [hunk 2]
```diff
@@ -9,4 +9,7 @@ poolConfigs:
     resourceConfigName: "resource_toolsalpha_v5"
   - poolId: "resource_toolsalpha_v6"
     size: 5
-    resourceConfigName: "resource_toolsalpha_v6"
\ No newline at end of file
+    resourceConfigName: "resource_toolsalpha_v6"
+  - poolId: "resource_toolsalpha_v7"
+    size: 5
+    resourceConfigName: "resource_toolsalpha_v7"
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply src/main/resources/config/toolsalpha/pool_schema.yml:15]: Done

File: src/main/resources/config/resource_schema.yaml [hunk 3]
```diff
@@ -66,6 +66,10 @@ components:
           $ref: '#/components/schemas/KubernetesEngine'
         serviceUsage:
           $ref: '#/components/schemas/ServiceUsage'
+        securityGroup:
+          description: |-
+            The security group to be used for this project
+          type: string
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply src/main/resources/config/resource_schema.yaml:72]: Thought about restricting it to an enum, but decided against it. 
reason - ease of maintenance. 

- Part of the reason we chose the type as a string, is that Leo / others can tomorrow add more specific allowLists. i.e the  allowLists can be named / renamed as <customName1>. If the value type here is an enum, we would likely need changes here as well as Leo, and UI if tomorrow we let this be configured by an admin. If its a string, No changes are needed in the buffer (we are just a pass-through)
- Unlikely scenario where we need to map client value to a different underlying value - easier with enum, but flexible with strings
- Enums need to be case-specific, lessening tight coupling on API where possible. Buffer can sanitize values where needed
- if client implementation does not support NULL values for enums, we risk the client inadvertently sending a default value.

not against enum per say, open to change if it adds value

File: src/main/java/bio/terra/buffer/service/resource/flight/GoogleProjectConfigUtils.java [hunk 4]
```diff
@@ -105,4 +106,14 @@ public static Optional<Long> bigQueryDailyUsageOverrideValueMebibytes(
     long value = bigQueryQuotas.getBigQueryDailyUsageQuotaOverrideValueMebibytes().longValue();
     return Optional.of(value);
   }
+
+  public static Optional<String> getSecurityGroup(GcpProjectConfig gcpProjectConfig) {
+    if (gcpProjectConfig.getSecurityGroup() != null) {
+      String secGroup = gcpProjectConfig.getSecurityGroup().trim();
+      if (!secGroup.isEmpty()) {
+        return Optional.of(secGroup);
+      }
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply src/main/java/bio/terra/buffer/service/resource/flight/GoogleProjectConfigUtils.java:115]: done, thanks!

File: src/test/java/bio/terra/buffer/service/resource/flight/GoogleProjectConfigUtilsTest.java [hunk 5]
```diff
@@ -1,22 +1,38 @@
...
+    GcpProjectConfig gcpProjectConfig = new GcpProjectConfig();
+    assertTrue(GoogleProjectConfigUtils.getSecurityGroup(gcpProjectConfig).isEmpty());
+
+    gcpProjectConfig.setSecurityGroup("   ");
+    assertTrue(GoogleProjectConfigUtils.getSecurityGroup(gcpProjectConfig).isEmpty());
+
+    gcpProjectConfig.setSecurityGroup("secGroup");
+    assertEquals(
+        gcpProjectConfig.getSecurityGroup(),
+        GoogleProjectConfigUtils.getSecurityGroup(gcpProjectConfig).orElse(Strings.EMPTY));
```
Comment (mikenev) [inline src/test/java/bio/terra/buffer/service/resource/flight/GoogleProjectConfigUtilsTest.java:36]: is the .orElse(Strings.EMPTY) needed here?

**Review by mikenev — APPROVED** [review]

File: src/test/java/bio/terra/buffer/service/resource/flight/GoogleProjectConfigUtilsTest.java (hunk 5 above)
Comment (dexamundsen) [reply src/test/java/bio/terra/buffer/service/resource/flight/GoogleProjectConfigUtilsTest.java:36]: Yes, it is needed. 
reason: function 'GoogleProjectConfigUtils.getSecurityGroup(gcpProjectConfig)' returns a optional<string> and the first param is a string. Need to get the string value from the Optional<> type.

**Pull request merged**